no-useless-rename has false negative

**What version of ESLint are you using?**

- v2.13.1

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

    $ echo "const {a, b: b} = obj" | eslint --stdin --env es6 --rule no-useless-rename:error --no-eslintrc

Or in our online demo:

![image](https://cloud.githubusercontent.com/assets/1937871/16256803/85b66bf8-388f-11e6-8c8c-110fd7bffb52.png)

**What did you expect to happen?**

```
<text>
  1:11  error  Destructuring assignment b unnecessarily renamed  no-useless-rename

✖ 1 problem (1 error, 0 warnings)
```

**What actually happened? Please include the actual, raw output from ESLint.**

No errors

--- a/lib/rules/no-useless-rename.js
+++ b/lib/rules/no-useless-rename.js
@@ -84,7 +84,7 @@ module.exports = {
 
             for (i = 0; i < properties.length; i++) {
                 if (properties[i].shorthand) {
-                    return;
+                    continue;
                 }
 
                 /**
@@ -94,7 +94,7 @@ module.exports = {
                  * so there is no "renaming" occurring here.
                  */
                 if (properties[i].computed || !properties[i].key) {
-                    return;
+                    continue;
                 }
 
                 if (properties[i].key.type === "Identifier" && properties[i].key.name === properties[i].value.name ||

--- a/tests/lib/rules/no-useless-rename.js
+++ b/tests/lib/rules/no-useless-rename.js
@@ -146,6 +146,12 @@ ruleTester.run("no-useless-rename", rule, {
             errors: ["Destructuring assignment foo unnecessarily renamed."]
         },
         {
+            code: "let {a, foo: foo} = obj;",
+            output: "let {a, foo} = obj;",
+            parserOptions: { ecmaVersion: 6 },
+            errors: ["Destructuring assignment foo unnecessarily renamed."]
+        },
+        {
             code: "let {foo: foo, bar: baz} = obj;",
             output: "let {foo, bar: baz} = obj;",
             parserOptions: { ecmaVersion: 6 },

